### PR TITLE
Add types

### PR DESCRIPTION
**Pull request by pd4d10**
Upgrade `hast-util-sanitize` to `^3.0.0` to use its types

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,15 +23,18 @@
...
     "Titus Wormer <tituswormer@gmail.com> (https://wooorm.com)"
   ],
+  "types": "types/index.d.ts",
   "files": [
+    "types/index.d.ts",
     "index.js"
   ],
   "dependencies": {
-    "hast-util-sanitize": "^2.0.0"
+    "hast-util-sanitize": "^3.0.0"
```
Comment (ChristianMurphy) [inline package.json:32]:
```
  types/index.d.ts:3:22

  ✖  3:22  unified should be listed in the project's dependencies. Run npm i -S unified to add it  import/no-extraneous-dependencies
```

https://travis-ci.org/github/rehypejs/rehype-sanitize/jobs/713293480#L233-L234

**Review by ChristianMurphy — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (pd4d10) [reply package.json:32]: OK. Shall we add it to `dependencies`? It seems more like a `peerDependencies`

Comment (ChristianMurphy) [reply package.json:32]: unified has primarily used `dependencies` for types directly used, and `devDependencies` for types only used for testing.
`peerDependencies` can be problematic due to inconsistent handling across dependency managers, and even within versions of the same dependency manager.

In this case I'd lean toward `dependencies`.

**Review by ChristianMurphy — APPROVED** [review]
Thanks @pd4d10! LGTM

**Comment by wooorm** [comment]
@ChristianMurphy As you approved, are we thus adding `unified` as a dep to all plugins? 🤔

**Comment by ChristianMurphy** [comment]
Most unified plugins use the `unified.Plugin` type.
We have some options, `dependencies` to have the typings automatically install, `peerDependencies` for a semi-manual install. `devDependencies` for a fully manual install and needing instructions.
Based off past discussions my understand is unified is moving towards including the needed types automatically (though `dependencies`)
Did/do you have a different preference/interpretation?

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
Thanks, released in 4.0.0!

**Pull request merged**